contracts/interchain-token/HyperliquidInterchainToken.sol

```diff
@@ -9,13 +9,16 @@ import { IHyperliquidDeployer } from '../interfaces/IHyperliquidDeployer.sol';
 /**
  * @title HyperliquidInterchainToken
  * @notice This contract implements an interchain token with Hyperliquid-specific modifications.
- * @dev Inherits from HyperLiquidDeployer first to ensure slot 0 is properly reserved,
+ * @dev Inherits from HyperLiquidDeployer to use keccak256-based storage slots for deployer addresses,
  * then from InterchainToken for standard functionality.
  * This maintains the standard InterchainToken while providing Hyperliquid compatibility.
  */
 contract HyperliquidInterchainToken is HyperliquidDeployer, InterchainToken, IHyperliquidDeployer {
     error NotService(address caller);
 
+    /// bytes32(uint256(keccak256('hyperliquid-interchain-token-deployer')) - 1)
+    bytes32 private constant CURRENT_DEPLOYER_SLOT = 0x8f802fa116fa5e7a7e0f94874e9214762d5a91faa240ab5140b3a79fbca28666;
```

Comment:
deployer logic should be encapsulated in the Deployer contract, not in the token contract, otherwise there's no point of the abstraction